### ISSUE TITLE
v2021.6.755784270 - TypeError: s.notebook.registerNotebookCellStatusBarItemProvider is not a function

### ISSUE THREAD

**Issue by MatiasDuhalde** [microsoft/vscode-jupyter#5537]
## Environment data

-   VS Code version: 1.56.0-insider
-   Jupyter Extension version (available under the Extensions sidebar): v2021.6.755784270
-   Python Extension version (available under the Extensions sidebar): v2021.3.680753044
-   OS (Windows | Mac | Linux distro) and version: Arch Linux 5.11.14-arch1-1
-   Python and/or Anaconda version: Python 3.9.3 64-bit
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): N/A
-   Jupyter server running: N/A

## Expected behaviour

Extension is successfully activated.

## Actual behaviour

Extension fails to activate, an error popup appears showing the following message:

`Extension activation failed, run the 'Developer: Toggle Developer Tools' command for more information. Source: Jupyter (Extension)`

Python extension still works, at least syntax highlighting, debugging, and code linting (via `pylint`).

## Steps to reproduce:

[**NOTE**: Self-contained, minimal reproducing code samples are **extremely** helpful and will expedite addressing your issue]

1. Open Visual Studio Code with the extension installed.
2. Open a `.py` file to trigger the Python extension, which in turn triggers the Jupyter extension.
3. The popup indicating the error should appear right after the previous step.

## Logs

<details>

<summary>Output for <code>Jupyter</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Jupyter</code>)
</summary>

<p>

```
User belongs to experiment group 'pythonJediLSP'
User belongs to experiment group 'jupyterTest'
User belongs to experiment group 'pythonDiscoveryModule'
User belongs to experiment group 'jupyterEnhancedDataViewer'
User belongs to experiment group 'NativeNotebookEditor'
Info 2021-04-17 00:23:13: ZMQ install verified.
Info 2021-04-17 00:23:13: Attempting to start a server because of preload conditions ...
Error 2021-04-17 00:23:13: Failed to register VS Code Notebook API [TypeError: s.notebook.registerNotebookCellStatusBarItemProvider is not a function
	at u.registerNotebookCellStatusBarItemProvider (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47:540799)
	at g.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:49:651504)
	at processTicksAndRejections (internal/process/task_queues.js:97:5)
	at async Promise.all (index 18)
	at async Promise.all (index 0)
	at async f.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47:290398)]
Error 2021-04-17 00:23:13: Failure during activation. [TypeError: s.notebook.registerNotebookCellStatusBarItemProvider is not a function
	at u.registerNotebookCellStatusBarItemProvider (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47:540799)
	at g.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:49:651504)
	at processTicksAndRejections (internal/process/task_queues.js:97:5)
	at async Promise.all (index 18)
	at async Promise.all (index 0)
	at async f.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47:290398)]
Error 2021-04-17 00:23:13: sendStartupTelemetry() failed. [TypeError: s.notebook.registerNotebookCellStatusBarItemProvider is not a function
	at u.registerNotebookCellStatusBarItemProvider (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47:540799)
	at g.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:49:651504)
	at processTicksAndRejections (internal/process/task_queues.js:97:5)
	at async Promise.all (index 18)
	at async Promise.all (index 0)
	at async f.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47:290398)]
Error 2021-04-17 00:23:13: extension activation failed [TypeError: s.notebook.registerNotebookCellStatusBarItemProvider is not a function
	at u.registerNotebookCellStatusBarItemProvider (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47:540799)
	at g.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:49:651504)
	at processTicksAndRejections (internal/process/task_queues.js:97:5)
	at async Promise.all (index 18)
	at async Promise.all (index 0)
	at async f.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47:290398)]
Error 2021-04-17 00:23:13: Failed to active the Jupyter Extension [TypeError: s.notebook.registerNotebookCellStatusBarItemProvider is not a function
	at u.registerNotebookCellStatusBarItemProvider (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47:540799)
	at g.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:49:651504)
	at processTicksAndRejections (internal/process/task_queues.js:97:5)
	at async Promise.all (index 18)
	at async Promise.all (index 0)
	at async f.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47:290398)]
Info 2021-04-17 00:23:14: Found existing keyfile at /home/tracert6/.config/Code - Insiders/User/globalStorage/ms-toolsai.jupyter/nbsecret
```

</p>
</details>

<details>

<summary>Developer tools console relevant output
</summary>

<p>

```
notificationsAlerts.ts:40 Extension activation failed, run the 'Developer: Toggle Developer Tools' command for more information.

console.ts:137 [Extension Host] extension activation failed TypeError: s.notebook.registerNotebookCellStatusBarItemProvider is not a function
	at u.registerNotebookCellStatusBarItemProvider (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47:540799)
	at g.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:49:651504)
	at processTicksAndRejections (internal/process/task_queues.js:97:5)
	at async Promise.all (index 18)
	at async Promise.all (index 0)
	at async f.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47:290398) (at /home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:32:774065)

console.ts:137 [Extension Host] stack trace: TypeError: s.notebook.registerNotebookCellStatusBarItemProvider is not a function	at u.registerNotebookCellStatusBarItemProvider (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47:540799)	at g.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:49:651504)	at processTicksAndRejections (internal/process/task_queues.js:97:5)	at async Promise.all (index 18)	at async Promise.all (index 0)	at async f.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47:290398) (at /opt/visual-studio-code-insiders/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:105:30533)

mainThreadExtensionService.ts:59 [[object Object]]s.notebook.registerNotebookCellStatusBarItemProvider is not a function

mainThreadExtensionService.ts:60 TypeError: s.notebook.registerNotebookCellStatusBarItemProvider is not a function
	at u.registerNotebookCellStatusBarItemProvider (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47)
	at g.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:49)
	at processTicksAndRejections (internal/process/task_queues.js:97)
	at async Promise.all (index 18)
	at async Promise.all (index 0)
	at async f.activate (/home/tracert6/.vscode-insiders/extensions/ms-toolsai.jupyter-2021.6.755784270/out/client/extension.js:47)
```

</p>
</details>

**Comment by MatiasDuhalde** [comment]
It should also be important to note that downgrading to version `v2021.6.751425733` (the previous one) fixes the problem. No errors are shown in the Output panel for the Jupyter extension.

**Comment by claudiaregio** [comment]
Hi @MatiasDuhalde, I am able to reproduce this issue. My guess is that this is related to a recent API change, will get it fixed as soon as possible.

**Comment by IanMatthewHuff** [comment]
@MatiasDuhalde Sorry about the issue here. In VS Code insiders can you run the "Check for Updates" command. I believe that it's in the top left menu, or available from the command palette. 

There was recently a breaking API change, and the error message that you are seeing here would indicate that you are on an older version of VS Code insiders. VS Code insiders updates daily so if you were a day or two behind then .registerNotebookCellStatusBarItemProvider would not have been available in your build. 

I'm confidant that this should be the issue. But if you are still seeing this error message and Code-insider indicates that there is no update available, then can you please share your Code-Insiders version info? It would need to be more than just the 1.56.0-insiders version number, as that just indicates the general version, not the daily build. The Help: About command should show the full build number and date of the version that you are on.

**Comment by claudiaregio** [comment]
@IanMatthewHuff 

I have no pending updates and still reproduce 

Version: 1.56.0-insider (user setup)
Commit: ce791d27dc9889a392f4d29abefe9c53eb0b5129
Date: 2021-04-16T11:25:09.662Z
Electron: 11.4.2
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.19042

**Comment by IanMatthewHuff** [comment]
@claudiaregio I'm going to loop with vscode team, see if we can get to the bottom of this. @MatiasDuhalde sorry about the issue here, seems like something is amis.

**Comment by MatiasDuhalde** [comment]
Hello @IanMatthewHuff and @claudiaregio , thanks for your comments! There was an update available in my case that for some reason didn't show up yesterday. The [AUR](https://aur.archlinux.org/packages/visual-studio-code-insiders-bin/) from where I installed the package for Arch seemed to be out of date. I updated the package to the latest version found in https://code.visualstudio.com/insiders/

More specifically I installed the Linux `tar.gz` file for 64 bit architectures, with filename `code-insider-x64-1618573217.tar.gz` and SHA256 `7f1aa38158ec5ee0cdb4a4025bcf383c4f776429e7d1c0df91ccb343bc28d1be`.

Now I get no errors with this version + Jupyter Extension latest version `v2021.6.755784270`, and the output log is clean.

My `code-insiders --version` output is now:
```
1.56.0-insider
ce791d27dc9889a392f4d29abefe9c53eb0b5129
x64
```
This is odd however, since the commit hash seems to be the same as @claudiaregio . Maybe there's a problem with the build for Windows 10?

**Comment by IanMatthewHuff** [comment]
@MatiasDuhalde Thanks for the info, and glad that you are unblocked. Yeah, I was very surprised when I saw this error, doesn't seem to me like something that should have been happening. You might be right that there is some type of build specific issue here. I'm on Mac myself, but I'm pretty sure that some of our other developers are on Windows and didn't see this issue when we released on Friday. Might need more info from the VS Code folks here.

Thanks for replying back and providing more info though, that will help us to diagnose this, and makes me less worried that this is a super widespread issue.
